hack/cluster-deploy.sh

```diff
@@ -17,29 +17,88 @@
 # Copyright 2022 The Kepler Contributors
 #
 
-set -ex pipefail
-
-source ./hack/common.sh
-
-CLUSTER_PROVIDER=${CLUSTER_PROVIDER:-kubernetes}
-MANIFESTS_OUT_DIR=${MANIFESTS_OUT_DIR:-"_output/generated-manifest"}
-
-function main() {
-    [ ! -d "${MANIFESTS_OUT_DIR}" ] && echo "Directory ${MANIFESTS_OUT_DIR} DOES NOT exists. Run make generate first."
-    
-    if [ "$CLUSTER_PROVIDER" == "microshift" ]
-    then
-        sed "s/localhost:5001/registry:5000/g" ${MANIFESTS_OUT_DIR}/deployment.yaml > ${MANIFESTS_OUT_DIR}/deployment.yaml.tmp && \
-            mv ${MANIFESTS_OUT_DIR}/deployment.yaml.tmp ${MANIFESTS_OUT_DIR}/deployment.yaml
-    fi
-    echo "Deploying manifests..."
-    # Ignore errors because some clusters might not have prometheus operator
-    echo "Deploying with image:"
-    cat ${MANIFESTS_OUT_DIR}/deployment.yaml | grep "image:"
-    kubectl apply -f ${MANIFESTS_OUT_DIR} || true
-    
-    # verify kepler deployment on cluster
-    ./hack/verify.sh kepler
+set -eu -o pipefail
+# constants
+PROJECT_ROOT="$(git rev-parse --show-toplevel)"
+declare -r CTR_CMD=${CTR_CMD:-docker}
+declare -r CLUSTER_PROVIDER=${CLUSTER_PROVIDER:-kind}
+declare -r MANIFESTS_OUT_DIR=${MANIFESTS_OUT_DIR:-"_output/generated-manifest"}
+
+declare IMAGE_REPO="${IMAGE_REPO:-localhost:5001}"
+declare IMAGE_TAG="${IMAGE_TAG:-devel}"
+declare NO_BUILD=false
+
+source "$PROJECT_ROOT/hack/utils.sh"
+
+build_kepler() {
+    header "Build Kepler"
+    $NO_BUILD && {
+        info "skipping building of kepler image"
+        return 0
+    }
+    run make build_containerized \
+        IMAGE_REPO="$IMAGE_REPO" \
+        IMAGE_TAG="$IMAGE_TAG" \
+        CTR_CMD="$CTR_CMD"
+}
+push_kepler() {
+    header "Push Kepler Image"
+    $NO_BUILD && {
+        info "skipping pushing of kepler image"
+        return 0
+    }
+    run make push-image \
+        IMAGE_REPO="$IMAGE_REPO" \
+        IMAGE_TAG="$IMAGE_TAG" \
+        CTR_CMD="$CTR_CMD"
+}
+
+cluster_prereqs() {
```

Comment:
This is not cluster pre-requisites right? This is Kepler pre-requisites.

Reply:
Yeah right more of how to deploy Kepler on cluster